File: FINAL_PROJECT/react/Rednav/src/components/Flights.js
import React from "react";
import FlightBoard from "./Flights/FlightBoard";

const Home = (props) => {
  const columns = [
    "airline",
    "origin country",
    "destination country",
    "departure time",
    "landing time",
    "remaining tickets",
  ];
  const flights = [
    {
      airline_company_id: 1,
      origin_country_id: 2,
      destination_country_id: 3,
      departure_time: "10:15",
      landing_time: "11:11",
      remaining_tickets: 22,
    },
    {
      airline_company_id: 2,
      origin_country_id: 3,
      destination_country_id: 4,
      departure_time: "14:32",
      landing_time: "20:22",
      remaining_tickets: 22,
    },
  ];
  return (
    <React.Fragment>
        <div className="container">
          <h4 className="center">Home</h4>
          <FlightBoard list={flights} colNames={columns} />
        </div>
    </React.Fragment>
  );
};

export default Home;

File: FINAL_PROJECT/tables/flights.py
from db_files.db_config import Base
from sqlalchemy.orm import backref, relationship
from sqlalchemy import Column, Integer, DateTime, BigInteger, ForeignKey


class Flights(Base):
    __tablename__ = 'flights'

    id = Column(BigInteger(),
                primary_key=True,
                autoincrement=True)
    airline_company_id = Column(BigInteger(),
                                ForeignKey(
                                'airline_companies.id',
                                ondelete='CASCADE'),
                                nullable=False)
    origin_country_id = Column(BigInteger(),
                               ForeignKey(
                               'countries.id',
                               ondelete='CASCADE'),
                               unique=False,
                               nullable=False)
    destination_country_id = Column(BigInteger(),
                                    ForeignKey(
                                    'countries.id',
                                    ondelete='CASCADE'),
                                    unique=False,
                                    nullable=False)
    departure_time = Column(DateTime(),
                            unique=False,
                            nullable=False)
    landing_time = Column(DateTime(),
                          unique=False,
                          nullable=False)
    remaining_tickets = Column(Integer())

    company = relationship("AirlineCompanies",

                           backref=backref(
                               "flights",
                               uselist=False,
                               passive_deletes=True))
    origin = relationship("Countries",
                          foreign_keys=[
                              origin_country_id],
                          backref=backref("oc_flights",
                                          uselist=False))
    destination = relationship("Countries",
                               foreign_keys=[
                                   destination_country_id],
                               backref=backref("dc_flights",
                                               uselist=False))

    def data_for_web(self):
        return {'id': self.id, 'airline_company': self.company.name,
                'origin_country': self.origin.name, "destination_country": self.destination.name, "departure_time": self.departure_time, "landing_time": self.landing_time, "remaining_tickets": self.remaining_tickets}

    def __repr__(self):
        return f'\n<Flight id={self.id} Airline Co. id={self.airline_company_id} Origin country id={self.origin_country_id} Desitnation country id={self.destination_country_id} Departure time={self.departure_time} Landing time={self.landing_time} Remaining tickets={self.remaining_tickets}>'

    def __str__(self):
        return f'<Flight id={self.id} Airline Co. id={self.airline_company_id} Origin country id={self.origin_country_id} Desitnation country id={self.destination_country_id} Departure time={self.departure_time} Landing time={self.landing_time} Remaining tickets={self.remaining_tickets}>'
